services/QuillLMS/engines/evidence/app/services/evidence/research/gen_ai/data_importer.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module Evidence
+  module Research
+    module GenAI
+      class DataImporter < ApplicationService
+        attr_reader :file_path, :file_name, :examples_path
+
+        def initialize(file_path:, file_name:, examples_path:)
+          @file_path = file_path
+          @file_name = file_name
+          @examples_path = examples_path
+        end
+
+        # rubocop:disable Metrics/CyclomaticComplexity
+        def run
+          passage_prompts.each do |passage_prompt|
+            conjunction = passage_prompt.conjunction
+            training_file_name = data['files'][conjunction]['train'].split('/').last
+            validation_file_name = data['files'][conjunction]['validation'].split('/').last
+
+            [training_file_name, validation_file_name].each do |examples_file_name|
+              break if passage_prompt.passage_prompt_responses.count >= 100
+
+              examples_file_abs_path = File.join(examples_path, examples_file_name)
+              next unless File.exist?(examples_file_abs_path)
+
+              File.open(examples_file_abs_path, 'r') do |file|
+                file.each_line do |line|
```

Comment:
If you want to get rid of the `Metrics/CyclomaticComplexity` you could make this block, e.g. `parse_line(line)` it's own method.

Reply:
When I transition this over to S3, this will go away.